### PR TITLE
add get user list functionality

### PR DESCRIPTION
**Pull request by angeltabar**

### CODE REVIEW

**Review by angelsanchez — CHANGES_REQUESTED** [review]

File: app/api/entity/error.go [hunk 1]
```diff
@@ -12,6 +12,9 @@ var (
 	// ErrUserNotFound error definition.
 	ErrUserNotFound = errors.New("user not found")
 
+	// ErrListUsersEmpty error definition.
+	ErrListUsersEmpty = errors.New("user list empty")
```
[the remark this answers is not in the document]
Comment (angeltabar) [reply app/api/entity/error.go]: removed

File: app/api/infrastructure/repository/user.go [hunk 2]
```diff
@@ -91,6 +91,37 @@ func (m *userMongoDBRepo) GetByEmail(ctx context.Context, email string) (entity.
 	return m.getByProp(ctx, "email", email)
 }
 
+// FindAll retrieves all the existing users.
+func (m *userMongoDBRepo) FindAll(ctx context.Context) ([]entity.User, error) {
+	m.logger.Debugf("Getting all users from \"%s\" collection...", userCollName)
+
+	cursor, err := m.collection.Find(ctx, bson.M{})
+	if err != nil {
+		if errors.Is(err, mongo.ErrNoDocuments) {
```
[the remark this answers is not in the document]
Comment (angeltabar) [reply app/api/infrastructure/repository/user.go]: that's true. Fix done

File: app/api/infrastructure/graph/schema.resolvers.go [hunk 3]
```diff
@@ -92,7 +92,12 @@ func (r *queryResolver) Project(ctx context.Context, id string) (*entity.Project
 }
 
 func (r *queryResolver) Users(ctx context.Context) ([]entity.User, error) {
-	panic(entity.ErrNotImplemented)
+	users, err := r.users.FindAll(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return users, nil
```
Comment (angelsanchez) [inline app/api/infrastructure/graph/schema.resolvers.go:100]: unnecessary code
```suggestion
return  r.users.FindAll(ctx)
```

**Review by angelsanchez — CHANGES_REQUESTED** [review]

File: app/api/infrastructure/graph/schema.resolvers.go (hunk 3 above)
Comment (angeltabar) [reply app/api/infrastructure/graph/schema.resolvers.go]: done!

**Pull request merged**